packages/announcement-controller/src/AnnouncementController.ts

```diff
@@ -1,96 +1,116 @@
+export type StateAnnouncementMap = {
   [id: number]: StateAnnouncement;
-}
-
-/**
- * AnnouncementConfig will hold the active announcements
- */
-export interface AnnouncementConfig extends BaseConfig {
-  allAnnouncements: AnnouncementMap;
-}
+};
 
 /**
  * Announcement state will hold all the seen and unseen announcements
  * that are still active
  */
-export interface AnnouncementState extends BaseState {
+export type AnnouncementState = {
   announcements: StateAnnouncementMap;
-}
+};
+
+export type AnnouncementControllerActions = GetAnnouncementState;
+export type AnnouncementControllerEvents = AnnouncementStateChange;
+
+export type GetAnnouncementState = {
+  type: `${typeof controllerName}:getState`;
+  handler: () => AnnouncementState;
+};
+
+export type AnnouncementStateChange = {
+  type: `${typeof controllerName}:stateChange`;
+  payload: [AnnouncementState, Patch[]];
+};
+
+const controllerName = 'AnnouncementController';
 
 const defaultState = {
   announcements: {},
 };
 
+const metadata = {
+  announcements: {
+    persist: true,
+    anonymous: false,
+  },
+};
+
+export type AnnouncementControllerMessenger = RestrictedControllerMessenger<
+  typeof controllerName,
+  AnnouncementControllerActions,
+  AnnouncementControllerEvents,
+  never,
+  never
+>;
+
 /**
  * Controller for managing in-app announcements.
  */
-export class AnnouncementController extends BaseController<
-  AnnouncementConfig,
-  AnnouncementState
+export class AnnouncementController extends BaseControllerV2<
+  typeof controllerName,
+  AnnouncementState,
+  AnnouncementControllerMessenger
 > {
   /**
    * Creates a AnnouncementController instance.
    *
-   * @param config - Initial options used to configure this controller.
-   * @param state - Initial state to set on this controller.
+   * @param args - The arguments to this function.
+   * @param args.messenger - Messenger used to communicate with BaseV2 controller.
+   * @param args.state - Initial state to set on this controller.
    */
-  constructor(config: AnnouncementConfig, state?: AnnouncementState) {
-    super(config, state || defaultState);
-    this.initialize();
-    this._addAnnouncements();
+  constructor({
+    messenger,
+    state,
+  }: {
+    messenger: AnnouncementControllerMessenger;
+    state?: AnnouncementState;
+  }) {
+    const mergedState = { ...defaultState, ...state };
+    super({ messenger, metadata, name: controllerName, state: mergedState });
+    this._addAnnouncements(mergedState.announcements);
   }
 
   /**
    * Compares the announcements in state with the announcements from file
    * to check if there are any new announcements
    * if yes, the new announcement will be added to the state with a flag indicating
    * that the announcement is not seen by the user.
+   *
+   * @param allAnnouncements - all announcements to
    */
-  private _addAnnouncements(): void {
+  private _addAnnouncements(allAnnouncements: StateAnnouncementMap): void {
```

Comment:
[We recently added a rule to our ESLint config](https://github.com/MetaMask/eslint-config/pull/244) that prohibits the use of `private` for private methods in TypeScript. Rather, we use the private fields feature. What do you think about:

``` suggestion
  #addAnnouncements(allAnnouncements: StateAnnouncementMap): void {
```

(note: this isn't the only required change, you'd also have to change where it gets called)